lib/BC/SleighLifter.cpp

```diff
@@ -714,28 +719,15 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
   }
 
 
-  std::optional<llvm::Value *>
-  GetResultOfBoolOpFunc(llvm::IRBuilder<> &bldr, OpCode opc,
-                        llvm::Value *in_lhs, llvm::Value *in_rhs) {
-    switch (opc) {
-      case CPUI_BOOL_AND: {
-        return bldr.CreateAnd(in_lhs, in_rhs);
-      }
-      case CPUI_BOOL_OR: {
-
-        return bldr.CreateOr(in_lhs, in_rhs);
-      }
-      case CPUI_BOOL_XOR: {
-        return bldr.CreateXor(in_lhs, in_rhs);
-      }
-      default: return std::nullopt;
-    }
-  }
   LiftStatus LiftBoolBinOp(llvm::IRBuilder<> &bldr, OpCode opc,
                            VarnodeData *outvar, VarnodeData lhs,
                            VarnodeData rhs) {
 
 
+    if (this->BOOL_BINARY_OPS.find(opc) == this->BOOL_BINARY_OPS.end()) {
```

Comment:
`if (is_supported(opc))` ? I assume all `OPS` collections have unique values wrt to each other? Then one method would nicely provide this functionality to all of them.

Reply:
Well we need to know if we want to try to lift the in param etc as a bool no?